include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -231,6 +231,36 @@ struct Session : Base<OrtSession> {
   TypeInfo GetOverridableInitializerTypeInfo(size_t index) const;
 };
 
+/**
+ * An experimental wrapper around the C++ API that provides support for modern C++ syntax/features
+ *   at the cost of some (minimal) memory overhead.
+ * Where possible default values are defined by the session's model metadata (i.e. input node names)
+ *   unless otherwise specified by the user.
+ * 
+ * NOTE: Experimental API components provide no guarantee of backwards compatibility in future releases.
+ */
+struct ExperimentalSession : Session {
+  ExperimentalSession(Env& env, ORTCHAR_T* model_path, SessionOptions& options) 
+    : Session(env, model_path, options) {};
+  ExperimentalSession(Env& env, void* model_data, size_t model_data_length, SessionOptions& options)
+    : Session(env, model_data, model_data_length, options) {};
+
+  // overloaded Run() with common defaults
+  std::vector<Value> Run(const Value& input_values, const RunOptions& run_options = RunOptions());
+  void Run(const Value& input_values, Value& output_values, const RunOptions& run_options = RunOptions());
```

Comment:
I would have expected something like std::vector<std::reference_wrapper<const Value>> or std::vector<const Value*>> so it's clear multiple values are being provided.

Reply:
@skottmckay I don't understand your comment. Doesn't a vector imply multiple values are output, whether or not it is a vector of Values, pointers to Values, or references to Values?

Reply:
I have modified the code so `input_values` and `output_values` are explicitly declared as std::vector now.